### PR TITLE
ci: lint and commitlint in its own workflow

### PR DESCRIPTION
**Pull request by jginsburgn**
Fixes https://github.com/karma-runner/karma-jasmine/issues/311

### CODE REVIEW

File: .github/workflows/lint.yml
Comment (devoto13) [inline .github/workflows/lint.yml:14]: IMO it is a waste to run a commit message validation/lint on every Node version. Do you expect cases when it would fail on one, but not the other? What do you think about making it a job running on the latest Node LTS in the `test.yml` instead?

Comment (devoto13) [inline .github/workflows/lint.yml:14]: IMO it is a waste to run a commit message validation/lint on every Node version. Do you expect cases when it would fail on one, but not the other? What do you think about making it a job running on the latest Node LTS in the `test.yml` instead?

File: .github/workflows/lint.yml
Comment (XhmikosR) [inline .github/workflows/lint.yml:10]: Consequently, I'd rename this to just `Lint` (and maybe the `linux` key above too)

File: .github/workflows/lint.yml
Comment (XhmikosR) [inline .github/workflows/lint.yml:30]: I'd look into removing commitlint if it's not used in the npm scripts and use an action instead.

Comment (devoto13) [reply .github/workflows/lint.yml:30]: Any particular benefits if we do so? The benefit of the script is that contributors can run it locally instead of pushing and waiting for the CI.

Comment (XhmikosR) [reply .github/workflows/lint.yml:30]: In that case forget about it :)

File: .github/workflows/lint.yml
Comment (jginsburgn) [reply .github/workflows/lint.yml:14]: I think it should instead run in the oldest Node.js we support (i.e. 10), not on the latest one. I think it is safer to assume that if it works on an older version it will work on a newer one than vice versa. WDYT?

**Comment by jginsburgn** [comment]
Tell me if I am wrong. @devoto13 wants linting (commit and source) to be done in a job under the `test.yml` workflow and @XhmikosR prefers it to be done in its own workflow (i.e. `lint.yml`). I can go with both. Let's just get to an agreement.

File: .github/workflows/lint.yml
Comment (XhmikosR) [reply .github/workflows/lint.yml:14]: It's a dev tool, I don't think we should care about old Node.js version, especially obsolete ones like Node.js 10.

File: .github/workflows/lint.yml
Comment (XhmikosR) [inline .github/workflows/lint.yml:23]: I wonder, should we get all the commits here? I mean, say you push a patch with the proper commit message, this would still flag older commits, but you can fix this with squash when merging.

Comment (jginsburgn) [reply .github/workflows/lint.yml:23]: I have proposed that we use rebase merging always. See https://github.com/karma-runner/karma/issues/3773.

File: .github/workflows/lint.yml
Comment (jginsburgn) [reply .github/workflows/lint.yml:10]: Let me do that!

**Comment by XhmikosR** [comment]
IMHO opinion commitlint shouldn't block the other tests, so the only way to achieve this is to have it in a separate workflow (separate workflow doesn't necessarily mean a separate workflow file BTW, although it wasn't possible to restart individual workflows until recently).

File: .github/workflows/lint.yml
Comment (jginsburgn) [reply .github/workflows/lint.yml:14]: Let's get rid of Node.js 10 in https://github.com/karma-runner/karma-jasmine/pull/297. SG?

File: .github/workflows/lint.yml
Comment (XhmikosR) [reply .github/workflows/lint.yml:23]: Rebase is good too, but if the patches all refer to one change, squash makes more sense IMHO.

**Comment by jginsburgn** [comment]
> IMHO opinion commitlint shouldn't block the other tests, so the only way to achieve this is to have it in a separate workflow (separate workflow doesn't necessarily mean a separate workflow file BTW, although it wasn't possible to restart individual workflows until recently).

I agree that it should not block other tests.

IIUC there is a one-to-one relation between a workflow and a file ([reference](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions#:~:text=Each%20workflow%20is%20stored%20as%20a%20separate%20YAML%20file%20in%20your%20code%20repository%2C%20in%20a%20directory%20called%20.github/workflows)).

Also, besides doing linting in its own workflow, we can do it in its own job and use `if: ${{ always() }}` for the other testing jobs to make sure they run regardless of the success status of the linting one ([reference](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#:~:text=In%20this%20example%2C%20job3%20uses%20the%20always()%20conditional%20expression%20so%20that%20it%20always%20runs%20after%20job1%20and%20job2%20have%20completed%2C%20regardless%20of%20whether%20they%20were%20successful)).

File: .github/workflows/lint.yml
Comment (jginsburgn) [reply .github/workflows/lint.yml:15]: [c825542](https://github.com/karma-runner/karma-jasmine/pull/313/commits/c8255429284effe5f74d9538c024ae87054d6cc4) done.

**Comment by XhmikosR** [comment]
Nowadays, you can restart single workflows, so we can just have one workflow file. I'll let you decide, I don't have strong feelings about it as long as the other tests are not blocked.

File: .github/workflows/lint.yml
Comment (jginsburgn) [reply .github/workflows/lint.yml]: [c825542](https://github.com/karma-runner/karma-jasmine/pull/313/commits/c8255429284effe5f74d9538c024ae87054d6cc4)

File: .github/workflows/lint.yml
Comment (jginsburgn) [reply .github/workflows/lint.yml:15]: In that case, squashing should be done in the head branch, prior to merging. Does that sg to you?

**Comment by jginsburgn** [comment]
> Nowadays, you can restart single workflows, so we can just have one workflow file. I'll let you decide, I don't have strong feelings about it as long as the other tests are not blocked.

I prefer its own file. Are you good with that @devoto13 ?

File: .github/workflows/lint.yml
Comment (jginsburgn) [reply .github/workflows/lint.yml]: @XhmikosR

File: .github/workflows/lint.yml
Comment (jginsburgn) [reply .github/workflows/lint.yml:15]: @XhmikosR

**Review by devoto13 — APPROVED** [review]
> Are you good with that @devoto13 ?

Everything LGTM

File: .github/workflows/lint.yml
Comment (jginsburgn) [reply .github/workflows/lint.yml:15]: Any other thoughts before merging? @XhmikosR

File: .github/workflows/lint.yml
Comment (jginsburgn) [reply .github/workflows/lint.yml]: Any other thoughts before merging? @XhmikosR

File: .github/workflows/lint.yml
Comment (XhmikosR) [reply .github/workflows/lint.yml:15]: Sorry, I don't have a lot of time nowadays :/ I'd say just proceed with it and drop Node.js 10 later. I still think we shouldn't use the minimum Node.js version here because it's a dev task, but I'll let you decide about it.

**Review by XhmikosR — APPROVED** [review]

**Pull request merged**
